dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -798,14 +840,121 @@ RtpsUdpDataLink::customize_queue_element(TransportQueueElement* element)
   return rtps;
 }
 
+TransportQueueElement*
+RtpsUdpDataLink::customize_queue_element_non_reliable_i(TransportQueueElement* element, bool requires_inline_qos, ResponseVec& responses, bool& deliver_after_send)
+{
+  const RepoId pub_id = element->publication_id();
+
+  RTPS::SubmessageSeq subm;
+
+  TransportSendElement* tse = dynamic_cast<TransportSendElement*>(element);
+  TransportCustomizedElement* tce =
+    dynamic_cast<TransportCustomizedElement*>(element);
+  TransportSendControlElement* tsce =
+    dynamic_cast<TransportSendControlElement*>(element);
+
+  Message_Block_Ptr data;
+
+  const ACE_Message_Block* msg = element->msg();
+
+  // Based on the type of 'element', find and duplicate the data payload
+  // continuation block.
+  if (tsce) {        // Control message
+    if (RtpsSampleHeader::control_message_supported(tsce->header().message_id_)) {
+      data.reset(msg->cont()->duplicate());
+      // Create RTPS Submessage(s) in place of the OpenDDS DataSampleHeader
+      RtpsSampleHeader::populate_data_control_submessages(
+                subm, *tsce, requires_inline_qos);
+    } else if (tsce->header().message_id_ == DATAWRITER_LIVELINESS) {
+      send_heartbeats_manual_i(tsce, responses);
+      deliver_after_send = true;
+      return 0;
+    } else {
+      element->data_dropped(true /*dropped_by_transport*/);
+      return 0;
+    }
+
+  } else if (tse) {  // Basic data message
+    // {DataSampleHeader} -> {Data Payload}
+    data.reset(msg->cont()->duplicate());
+    const DataSampleElement* dsle = tse->sample();
+    // Create RTPS Submessage(s) in place of the OpenDDS DataSampleHeader
+    RtpsSampleHeader::populate_data_sample_submessages(
+              subm, *dsle, requires_inline_qos);
+
+  } else if (tce) {  // Customized data message
+    // {DataSampleHeader} -> {Content Filtering GUIDs} -> {Data Payload}
+    data.reset(msg->cont()->cont()->duplicate());
+    const DataSampleElement* dsle = tce->original_send_element()->sample();
+    // Create RTPS Submessage(s) in place of the OpenDDS DataSampleHeader
+    RtpsSampleHeader::populate_data_sample_submessages(
+              subm, *dsle, requires_inline_qos);
+
+  } else {
+    return element;
+  }
+
+#ifdef OPENDDS_SECURITY
+  {
+    GuardType guard(this->strategy_lock_);
+    if (this->send_strategy_) {
+      send_strategy()->encode_payload(pub_id, data, subm);
+    }
+  }
+#endif
+
+  Message_Block_Ptr hdr(submsgs_to_msgblock(subm));
+  hdr->cont(data.release());
+  RtpsCustomizedElement* rtps =
```

Comment:
Doesn't seem like we need a named local variable for this, it can just be returned